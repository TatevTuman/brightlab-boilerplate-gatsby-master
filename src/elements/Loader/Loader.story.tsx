import React from 'react'
import { Meta, Story } from '@storybook/react/types-6-0'
import Loader, { LoaderProps } from './Loader'
import { storybook, colors } from '@utils'

interface LoaderStoryTemplateProps extends LoaderProps {}

const loaderTypes = [
  'Audio',
  'BallTriangle',
  'Bars',
  'Circles',
  'Grid',
  'Hearts',
  'MutatingDots',
  'None',
  'NotSpecified',
  'Oval',
  'Plane',
  'Puff',
  'RevolvingDot',
  'Rings',
  'TailSpin',
  'ThreeDots',
  'Triangle',
  'Watch'
]

export default {
  title: 'Elements/Loader',
  argTypes: {
    type: { control: storybook.controls.select(loaderTypes), name: 'Type' },
    width: { control: 'number', name: 'Width' },
    height: { control: 'number', name: 'Height' },
    radius: { control: 'number', name: 'Radius' },
    color: { control: storybook.controls.color, name: 'Color' },
    secondaryColor: { control: storybook.controls.color, name: 'SecondaryColor' },
    timeout: { control: 'number', name: 'Timeout' },
    style: { control: 'object', name: 'Style' },
    className: storybook.args.disabled
  }
} as Meta

const LoaderStoryTemplate: Story<LoaderStoryTemplateProps> = args => {
  return <Loader {...args} />
}

LoaderStoryTemplate.args = {
  type: loaderTypes[9] as 'Oval',
  width: 200,
  height: 200,
  radius: 0,
  color: colors[0],
  secondaryColor: colors[1],
  timeout: 100000,
  style: {}
}

export const LoaderStory = LoaderStoryTemplate.bind({})
LoaderStory.args = {
  ...LoaderStoryTemplate.args
}
