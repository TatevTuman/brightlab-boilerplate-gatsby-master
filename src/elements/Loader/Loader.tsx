import React, { CSSProperties } from 'react'
import LoaderComponent from 'react-loader-spinner'
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css'

type LoaderTypes =
  | 'Audio'
  | 'BallTriangle'
  | 'Bars'
  | 'Circles'
  | 'Grid'
  | 'Hearts'
  | 'MutatingDots'
  | 'None'
  | 'NotSpecified'
  | 'Oval'
  | 'Plane'
  | 'Puff'
  | 'RevolvingDot'
  | 'Rings'
  | 'TailSpin'
  | 'ThreeDots'
  | 'Triangle'
  | 'Watch'

export interface LoaderProps {
  color?: string
  height?: number
  radius?: number
  secondaryColor?: string
  timeout?: number // in milliseconds
  type?: LoaderTypes
  width?: number
  className?: string
  style?: CSSProperties
  page?: boolean
  layer?: boolean
}

const TIMEOUT = 100000

const Loader: React.FC<LoaderProps> = props => {
  const { className, style, color, secondaryColor, page, layer, ...otherProps } = props

  return (
    <div className={'loader ' + className} style={style} data-testid={'loader'} data-page={page} data-layer={layer}>
      <LoaderComponent {...otherProps} color={color} secondaryColor={secondaryColor} />
    </div>
  )
}

Loader.defaultProps = {
  type: 'Oval',
  color: 'text-black-1000',
  secondaryColor: 'text-white',
  height: 60,
  width: 60,
  timeout: TIMEOUT
}

export default Loader
