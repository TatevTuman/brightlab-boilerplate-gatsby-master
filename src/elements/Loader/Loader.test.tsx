import React from 'react'
import { act, render, cleanup, waitFor } from '@testing-library/react'
import Loader from './Loader'

jest.useFakeTimers()

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Loader', () => {
  const props = {
    type: 'Triangle' as const,
    color: 'primary',
    height: 60,
    width: 60,
    timeout: 15000
  }

  it('renders correctly', async () => {
    const { container } = render(<Loader {...props} />)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })

  it('renders color correctly', async () => {
    const { container } = render(<Loader {...props} />)
    const polygon = container.querySelector('polygon')
    expect(polygon).toHaveAttribute('stroke', 'var(--primary)')
  })

  it('renders size correctly', async () => {
    const { container } = render(<Loader {...props} />)
    const svg = container.querySelector('svg')
    expect(svg).toHaveAttribute('width', props.width.toString())
    expect(svg).toHaveAttribute('height', props.height.toString())
  })
})
