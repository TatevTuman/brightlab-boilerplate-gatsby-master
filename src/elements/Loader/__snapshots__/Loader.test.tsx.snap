// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
<div>
  <div
    class="loader undefined"
    data-testid="loader"
  >
    <div
      aria-busy="true"
      class=""
    >
      <div
        class="react-spinner-loader-svg"
      >
        <svg
          aria-label="audio-loading"
          height="60"
          id="triangle"
          viewBox="-3 -4 39 39"
          width="60"
        >
          <polygon
            fill="transparent"
            points="16,0 32,32 0,32"
            stroke="var(--primary)"
            stroke-width="1"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
